Docsp 33189 search index management (#456)
mongodb/docs-java: source/fundamentals/indexes.txt

Reviewer: The search index commands operate asynchronously, returning from the server before the index is confirmed to be successfully updated, created, or dropped. To determine when an index has been created or updated, users should run the listSearchIndexes command periodically until the desired changes appear. It might be beneficial to include this information in the documentation as well. Let me know what you think

Author: Quick question:
Is this the case for all index management by the driver or only for Atlas Search indexes? I was thinking of putting this information in the Operational Considerations section if it applies to all index management by the driver.

Reviewer: This is only the case for Atlas Search Indexes.

When you execute a query against MongoDB, your command can include various elements:

- Query criteria that specify fields and values you are looking for

> - Options that affect the query's execution such as read concern

Reviewer 2: ```suggestion
- Options that affect the query's execution, such as read concern
```

- Projection criteria to specify the fields MongoDB returns (optional)

> - Sort criteria to specify the order documents returned from MongoDB (optional)

Reviewer 2: ```suggestion
- Sort criteria to specify the order of documents returned from MongoDB (optional)
```

      name_1_age_-1

   MongoDB uses this index when you sort your data by either:

   - ``name`` ascending, ``age`` descending
   - ``name`` descending, ``age`` ascending

   Specifying a sort order of ``name`` and :guilabel:`age` ascending or :guilabel:`name` and ``age``
   descending requires an in-memory sort.

For more information on how to ensure your index covers your query criteria and projection, see the MongoDB manual
articles on :manual:`query coverage </core/query-optimization/#covered-query>`.

Operational Considerations
~~~~~~~~~~~~~~~~~~~~~~~~~~

To improve query performance, build indexes on fields that appear often in your application's queries and operations

> that return sorted results. Each index that you add consumes disk space and memory when active so you can track index
> memory and disk usage for capacity planning. In addition, when a write operation updates an indexed field, MongoDB

Reviewer 2: S: I think changing the should to can changed the meaning of this sentence. Could we say something like

```suggestion
that return sorted results. Each index that you add consumes disk space and memory when active, so we recommend that you track index
memory and disk usage for capacity planning. In addition, when a write operation updates an indexed field, MongoDB
```

``listSearchIndexes()`` (TODO: add API link) method to return the Atlas
Search indexes of a collection.

The following code example shows how to print a list of the search indexes of
a collection:

.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
   :language: java
   :dedent:
   :start-after: start list-search-indexes
   :end-before: end list-search-indexes

Update a Search Index
+++++++++++++++++++++

You can use the
``updateSearchIndex()`` (TODO: add API link)
method to update an Atlas Search index.

The following code shows how to update a search index:

.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
   :language: java
   :dedent:
   :start-after: start update-search-index
   :end-before: end update-search-index

Drop a Search Index
+++++++++++++++++++

You can use the
``dropSearchIndex()`` (TODO: add API link)
method to remove an Atlas Search index.

The following code shows how to delete a search index from a collection:

.. literalinclude:: /includes/fundamentals/code-snippets/SearchIndexMethods.java
   :language: java
   :dedent:
   :start-after: start drop-search-index
   :end-before: end drop-search-index

> 

Reviewer: The search index commands operate asynchronously, returning from the server before the index is confirmed to be successfully updated, created, or dropped. To determine when an index has been created or updated, users should run the `listSearchIndexes` command periodically until the desired changes appear. It might be beneficial to include this information in the documentation as well. Let me know what you think

.. tip::

>    Text indexes differ from the more fully-featured MongoDB Atlas Search

Reviewer 2: S: change the wording of "more fully-featured" to be less redundant.

Author: I'll replace it with the text from https://www.mongodb.com/docs/manual/core/indexes/index-types/index-text/
